flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/BuiltInFunctionTestBase.java

```diff
@@ -342,18 +286,53 @@ TestSpec testResult(
             return this;
         }
 
+        Stream<TestCase> getTestCases(Configuration configuration) {
+            return testItems.stream().map(testItem -> getTestCase(configuration, testItem));
+        }
+
+        private TestCase getTestCase(Configuration configuration, TestItem testItem) {
+            return new TestCase(
+                    testItem.toString(),
+                    () -> {
+                        final TableEnvironmentInternal env =
```

Comment:
I don't remember but is there an actual reason for `TableEnvironmentInternal` otherwise let's expose only `TableEnvironment`.

Reply:
It's needed to access the catalog manager to manually run the type inference, check line 353 of this class